drivers/stm32_eth/stm32_eth.c

```diff
@@ -42,6 +52,12 @@ static void _isr(netdev_t *netdev) {
     if(stm32_eth_get_rx_status_owned()) {
         netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
     }
+#if IS_USED(MODULE_STM32_ETH_LINK_UP)
+    if (link_status && first_call) {
+        netdev->event_callback(netdev, NETDEV_EVENT_LINK_UP);
+        first_call = 0;
+    }
+#endif
```

Comment:
...

by doing that, you can remove the `#ifdef` here

```suggestion
    if (IS_USED(MODULE_STM32_ETH_LINK_UP) && link_status && first_call) {
        netdev->event_callback(netdev, NETDEV_EVENT_LINK_UP);
        first_call = 0;
    }
```

Reply:
Why is `first_call` being set to `0` here?
Can `link_status` ever become `0` again?

Reply:
`first_call` is set to `0` here, so that `NETDEV_EVENT_LINK_UP` is not sent with every isr call because otherwise `dhcp_start` would be called on every event. 
Currently, `link_status` can never be `0` again, but it could be used for a link down-event in later use cases.